65black/StackOverflow-lite
# #159958224 Add User Signup and Login with JWT

What does this PR do?
Adds logic for user signup and login routes. These routes are protected with JWT.

Description of Task to be completed?
Have the following endpoints working:

POST api/v1/auth/signup
POST api/v1/auth/login

How should this be manually tested?
After cloning the repo, cd into it and run npm run start-dev.

For /api/v1/auth/signup
Then open Postman and make a POST request with the following data (Make sure you select x-www-form-urlencoded:

name: <your_name>
email: <email>
password: <your_password>
Copy the token from the response to be used for the login route.

For /api/v1/auth/login
Add the token you copied from the signup response to your header.

Then open Postman and make a POST request with the following data (Make sure you select x-www-form-urlencoded:

email: <email>
password: <your_password>
Note
You should change the connection to your own working database in the db.js file. This is because the database currently there is local.

What are the relevant pivotal tracker stories?
#159958224
